api/middleware/roleTicketsMiddleware.js

```diff
@@ -0,0 +1,36 @@
+const RoleTicket = require('../roleTickets/roleTicketsModel');
+const roleTicketSchema = require('../../data/schemas/roleTicketSchema');
+
+async function checkRoleTicketIdExists(req, res, next) {
+  const { role_ticket_id } = req.params;
+  const badIdErr = {
+    status: 404,
+    message: `Role Ticket with ID ${role_ticket_id} not found!`,
+  };
+  try {
+    const roleTicket = await RoleTicket.findByRoleTicketById(role_ticket_id);
+    if (!roleTicket) {
+      next(badIdErr);
+    }
+    req.roleTicket = roleTicket;
+    next();
+  } catch (err) {
+    next();
+  }
+}
+
+async function validateRoleTicket(req, res, next) {
+  try {
+    const payload = req.body;
+    const validatedRoleTicket = await roleTicketSchema.validate(payload);
+    req.roleTicket = validatedRoleTicket;
```

Comment:
Since `req.roleTicket` can also be set by the above middleware as well, it might be better to rename this variable to something like `req.roleTicketInput` to allow them to co-exist in future middlewares in the event they are needed. Right now, both middlewares "compete" for the variable, and whichever one runs last wins.

Reply:
Thanks for the suggestion. I have renamed the variable to req.roleTicketInput